Source/RMG/UserInterface/Widget/RomBrowserWidget.cpp

```diff
@@ -664,22 +746,39 @@ void RomBrowserWidget::on_RomBrowserThread_Finished(bool canceled)
     }
 
     // retrieve column settings
-    std::vector<int> columnVisibility = CoreSettingsGetIntListValue(SettingsID::RomBrowser_ColumnVisibility);
-    std::vector<int> columnOrder = CoreSettingsGetIntListValue(SettingsID::RomBrowser_ColumnOrder);
     std::vector<int> columnSizes = CoreSettingsGetIntListValue(SettingsID::RomBrowser_ColumnSizes);
+    std::vector<int> columnOrder = CoreSettingsGetIntListValue(SettingsID::RomBrowser_ColumnOrder);
+    std::vector<int> columnVisibility = CoreSettingsGetIntListValue(SettingsID::RomBrowser_ColumnVisibility);
 
     // temporarily disable stretching last column in list view
     this->listViewWidget->horizontalHeader()->setStretchLastSection(false);
 
-    // reset column visibility setting in config file if number of values is incorrect
-    if (!columnVisibility.empty() &&
-        columnVisibility.size() != this->listViewModel->columnCount())
+    // reset column sizes setting in config file if number of values is incorrect
```

Comment:
Why were these changes made?

Reply:
I already alluded to this in my initial changelog, but I changed `on_listViewWidget_sectionResized` and `on_RomBrowserThread_Finished` to prevent this from happening:
![RMG_Bug](https://user-images.githubusercontent.com/70489593/222929525-bf607ef9-857b-473b-8586-5f4047f9c3d6.png)
Before I changed it, the code to resize columns to their contents in `on_RomBrowserThread_Finished` skipped over columns that were hidden by default. When these columns were set to be visible, they didn't take the length of the header label into account when determining their sizes for the first time. For most of the columns, that wasn't an issue, but for columns whose header labels were longer than any of their contents, the label got cut off.

To correct this, I reorganized the code in `on_RomBrowserThread_Finished` so that the column sizes are calculated and set before visibility. This also allowed me to add an extra if statement to `on_listViewWidget_sectionResized` that greatly reduces the amount of code that needs to run when resetting column sizes in `on_RomBrowserThread_Finished` and the new `on_Action_ResetColumnSizes` function.